addons/project/models/project.py

```diff
@@ -339,6 +381,18 @@ def _compute_alias_value(self):
             else:
                 project.alias_value = "%s@%s" % (project.alias_name, project.alias_domain)
 
+    @api.depends('allow_project_sharing', 'access_url')
+    def _compute_public_link(self):
+        base_url = self.env[self._name].get_base_url()
+        if not base_url:
+            # Remove all public_link if it is already set in projects
+            self.filtered('public_link').public_link = ''
```

Comment:
?
```suggestion
            self.write({'public_link': False})
```

Reply:
It was to update only the ones with public_link set. 🙂 
But yeah, I can do a write to set all records in the `self` without filtering.

Reply:
I saw I do not return to stop the process of this method, after the line where you proposed to do a write. So, the compute unnecessarily continues its process. 😬